File: practices/websocket/src/main/java/com/grizz/wooman/websocket/config/MappingConfig.java
package com.grizz.wooman.websocket.config;

import com.grizz.wooman.websocket.handler.ChatWebSocketHandler;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.reactive.handler.SimpleUrlHandlerMapping;
import org.springframework.web.reactive.socket.WebSocketHandler;

import java.util.Map;

@Configuration
public class MappingConfig {
    @Bean
    SimpleUrlHandlerMapping simpleUrlHandlerMapping(
            ChatWebSocketHandler chatWebSocketHandler
    ) {
        Map<String, WebSocketHandler> urlMapper = Map.of(
                "/chat", chatWebSocketHandler
        );

        SimpleUrlHandlerMapping mapping = new SimpleUrlHandlerMapping();
        mapping.setOrder(1);
        mapping.setUrlMap(urlMapper);

        return mapping;
    }
}

File: practices/coroutine-mongo-chat/src/main/kotlin/com/grizz/wooman/coroutine/config/MappingConfig.kt
package com.grizz.wooman.coroutine.config

import com.grizz.wooman.coroutine.handler.ChatCoroutineWebSocketHandler
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.reactive.handler.SimpleUrlHandlerMapping
import java.util.Map

@Configuration
class MappingConfig {
    @Bean
    fun simpleUrlHandlerMapping(
        chatWebSocketHandler: ChatCoroutineWebSocketHandler
    ): SimpleUrlHandlerMapping {
        val urlMapper = Map.of("/chat", chatWebSocketHandler)
        return SimpleUrlHandlerMapping().also {
            it.order = 1
            it.urlMap = urlMapper
        }
    }
}